pact/provider/pact-broker-loader/src/main/java/org/arquillian/algeron/pact/provider/core/loader/pactbroker/PactBrokerLoader.java

```diff
@@ -80,13 +80,17 @@ public String getName() {
 
     @Override
     public List<URI> retrieve() throws IOException {
-        List<URI> pacts = new ArrayList<>();
-        for (String tag: Arrays.stream(pactBroker.tags())
+        return Arrays.stream(pactBroker.tags())
                 .map(tag -> getResolvedValue(tag))
-                .collect(toList())) {
-            pacts.addAll(loadPactsForProvider(providerName, tag));
-        }
-        return pacts;
+                .map(tag -> {
+                    try {
+                        return loadPactsForProvider(providerName, tag);
+                    } catch (IOException e) {
+                        throw new IllegalStateException(e);
```

Comment:
Do we really have to catch it here if retrieve() throws IOException anyway? Is it forced to check for this exception due to lambda? (not looking in the IDE so this can be stupid question :P)